src/ORM/DataQuery.php

```diff
@@ -196,9 +196,10 @@ public function setQueriedColumns($queriedColumns)
      * Ensure that the query is ready to execute.
      *
      * @param array|null $queriedColumns Any columns to filter the query by
+     * @param bool $preferConditions Indicates that conditions should be preferred over joins (used with "EXISTS")
      * @return SQLSelect The finalised sql query
      */
-    public function getFinalisedQuery($queriedColumns = null)
+    public function getFinalisedQuery($queriedColumns = null, $preferConditions = false)
```

Comment:
I may be a bit blind, but I see $preferConditions arg added, but I don't see any logic that depends on it. Is there some other part of the code that respects this value?

Reply:
If core ignores this value, could I argue it shouldn't be necessary?

Reply:
It's given to `augmentSQL`. I think it's reasonable that it can be added to core functionality too but I can't think of a use-case. 

It's difficult to gain detail about the query from extensions when using `limit` or `exists` so this is an indicator of that. Perhaps `preferConditions` is the wrong name.

Reply:
I'm moving this change to a new PR. I'll reference this feedback in it when I raise it.

Reply:
I've decided to drop it - thanks for prompting me to think this through a little more 😅